### DISCUSSION TITLE
Cloudflare KB Article Submission

### DISCUSSION THREAD

**Discussion by th3jesta** [firefly-iii/firefly-iii#7293] [Firefly III]
### Description

I would like to submit a short article to help those who have Firefly hosted behind Cloudflare optimizations. I have seen a few closed, unresolved issues from others with the same problem that I was able to solve.

### Problem
Certain UI elements will not load or function if you have specific Cloudflare optimizations enabled for your domain:
* Accordion menus
* Email address in the top-right
* Graphs
* Menu buttons
* etc.

### Solution
1. Log into your Cloudflare dashboard, click on **Rules** in the left menu, then **Configuration Rules**.
2. Click the **Create Rule** button, give it a name, and select **Custom filter expression** where:
        HOSTNAME    EQUALS    <firefly.yourdomain.com>
3. Scroll down to **Email Obfuscation** and **Rocket Loader**, add them, and make sure the toggles are in the disabled position. 
4. Save, wait a minute, and your Firefly app should be loading normally.

#### Additional notes
I discovered this issue by looking at my browser dev tools and seeing that the rocket-loader.js and email obfuscation script were blocked in the Firefly app. You may have more or fewer things blocked depending on your Cloudflare settings. Luckily, you can use the above method to disable any other Cloudflare features you may have enabled globally across your domains according to the rules and **Custom filter expressions** you set.

**Comment by th3jesta** [comment]
#2308 #2198

**Comment by JC5** [comment]
Interesting idea. The demo site is also behind cloudflare and I believe it works without these changes. I'll double check and get back to you.

**Comment by th3jesta** [comment]
I suspect you just don't have those features enabled globally or at all for the domains in Cloudflare.

**Reply by JC5** [reply]
Indeed, I don't. I'll add something to the FAQ.

**Comment by orlandoflv** [comment]
Thank you! this was very helpful

**Comment by Mega61** [comment]
Absolute banger to find this here!

Using Zoraxy and cloudfare this changes allowed me to fix the HTTPS deployment!
